### PR TITLE
break up the Poste::Template.render and .initialize methods to make it more composable

### PR DESCRIPTION
**Pull request by Hamms**
In preparation for switching it over to use ActionView for the rendering logic.

## Links

- [jira](https://codedotorg.atlassian.net/browse/FND-1020)

## Testing story

I'm relying on our [new email rendering test](https://github.com/code-dot-org/code-dot-org/pull/33362) to ensure this refactor doesn't change anything.

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: lib/cdo/poste.rb [hunk 1]
```diff
@@ -110,41 +110,69 @@ def self.unsubscribe(email, hashed_email, params={})
...
 
   class Template
-    def initialize(body, engine=TextRender::MarkdownEngine)
+    def initialize(path)
+      body = IO.read(path)
+
+      @engine = {
+        '.html' => TextRender::ErbEngine,
+        '.md' => TextRender::MarkdownEngine,
+      }[File.extname(path).downcase]
```
Comment (islemaster) [inline lib/cdo/poste.rb:120]: What happened to haml, txt, and yml?

Comment (Hamms) [reply lib/cdo/poste.rb:120]: Oh, I should keep haml in there; but txt and yml aren't actually supported extensions: https://github.com/code-dot-org/code-dot-org/blob/4c2808a6796d98b8d855a1e83a03be827be59054/lib/cdo/poste.rb#L77-L79

File: lib/cdo/poste.rb [hunk 2]
```diff
@@ -110,41 +110,70 @@ def self.unsubscribe(email, hashed_email, params={})
...
         params.merge! form.processed_data
         params['form'] = form
       end
       locals = OpenStruct.new(params).instance_eval {binding}
 
-      header = @header.result(locals) unless @header.nil?
+      header = render_header(locals)
+
       # TODO(andrew): Fix this so that we get a signal as to how often this is happening.
       # For more information, see https://www.pivotaltracker.com/story/show/104750788.
```
Comment (islemaster) [inline lib/cdo/poste.rb]: Looks like you've removed a guard clause here (hopefully unnecessary now).  Should this TODO also be removed? I'm pretty sure that's what it was referring to.

**Review by islemaster — APPROVED** [review]
:+1: LGTM

File: lib/cdo/poste.rb (hunk 2 above)
Comment (Hamms) [reply lib/cdo/poste.rb]: oh, good call. I don't have insight into whether or not the TODO would still be useful; do you have context for our use of this litmus tracking feature, or do you know who might?

Comment (islemaster) [reply lib/cdo/poste.rb]: :joy: Maybe?  Looks like it was introduced in November 2014 by @geoffrey-elliott in [this commit named "Fix conflicts."](https://github.com/code-dot-org/code-dot-org/commit/19e5aa707d01614ae2473b092fd19629eae03439#diff-fe80c20e87b389e81c85fe8da32b3afcR75) (If he's watching GitHub maybe he can shed some light on the situation.)

I checked out `emltrk.com` (which was blocked by uBlock, by the way) and Litmus does seem to still be a working service.  I have credentials in LastPass.  It looks like we've been on a "free account" for more than three years now, which doesn't have any features except for the ability to start paying again.

It also looks like we've been using the same tracking id in our HOC emails for years.

I think we can safely tear this out.  Might check with #marketing _just_ to be sure first.

Comment (Hamms) [reply lib/cdo/poste.rb]: Thanks for the sleuthing!

4136b77b16937f3c47c061b9f1625489

I'll take out this TODO and add a work item to track tearing out the tracking logic

Comment (islemaster) [reply lib/cdo/poste.rb]: Addressing in https://github.com/code-dot-org/code-dot-org/pull/33402.

**Pull request merged**